lib/blocks.php

```diff
@@ -147,6 +150,45 @@ function gutenberg_reregister_core_block_types() {
 
 add_action( 'init', 'gutenberg_reregister_core_block_types' );
 
+/**
+ * Registers block styles for a core block.
+ *
+ * @param string $block_name The block-name.
+ *
+ * @return void
+ */
+function gutenberg_register_core_block_styles( $block_name ) {
+	if ( ! current_theme_supports( 'split-block-styles' ) ) {
+		return;
+	}
+	$block_name = str_replace( 'core/', '', $block_name );
```

Comment:
We definitely can replicate `is_rtl` handling inside `register_block_type_from_metadata`. Still, it's fine to start with what this PR proposes 👍 

Maybe we could also reuse `register_block_style_handle` for now, this is how it looks like:

https://github.com/WordPress/wordpress-develop/blob/850e31dc58b2075a347fd2dab0eb62b21f2077b3/src/wp-includes/blocks.php#L144-L163

Reply:
I agree, `register_block_style_handle` does indeed seem like the right place to add these things. :+1: 
If this PR gets merged the changes should probably be ported to wp-core in a subsequent PR, and the `gutenberg_register_core_block_styles` function removed.